micom/workflows/build.py

```diff
@@ -186,7 +194,20 @@ def build_database(
     meta["id"] = meta.index
     meta["summary_rank"] = rank
 
-    if compress:
+    # compress is ignored if outpath does not end with ".zip"
+    if out_path.endswith(".zip"):
+        compressdict = {
+            False: zipfile.ZIP_STORED,
+            "zlib": zipfile.ZIP_DEFLATED,
+            "bz2": zipfile.ZIP_BZIP2,
+            "lzma": zipfile.ZIP_LZMA,
+        }
+        compress = compressdict.get(compress, "unknown")
+        # Check compression parameters before heavy computations
+        zipfile._check_compression(compress)
```

Comment:
Raise error above, don't depend on internal (unexported) functions.

Reply:
I added a manual error, but I kept the `zipfile._check_compression()` because I think it is useful because it checks if the relevant dependencies are installed for the compression method.

```python
Signature: zipfile._check_compression(compression)
Docstring: <no docstring>
Source:   
def _check_compression(compression):
    if compression == ZIP_STORED:
        pass
    elif compression == ZIP_DEFLATED:
        if not zlib:
            raise RuntimeError(
                "Compression requires the (missing) zlib module")
    elif compression == ZIP_BZIP2:
        if not bz2:
            raise RuntimeError(
                "Compression requires the (missing) bz2 module")
    elif compression == ZIP_LZMA:
        if not lzma:
            raise RuntimeError(
                "Compression requires the (missing) lzma module")
    else:
        raise NotImplementedError("That compression method is not supported")
```

If we remove it, the same error will be raised later by `zipfile.Zipfile()` on L223, but after the heavy `workflow()` computation (L220).

Your call :) .